File: PiwikTracker/PiwikDebugDispatcher.m
//
//  PiwikDebugDispatcher.m
//  PiwikTracker
//
//  Created by Mattias Levin on 29/08/14.
//  Copyright (c) 2014 Mattias Levin. All rights reserved.
//

#import "PiwikDebugDispatcher.h"


@implementation PiwikDebugDispatcher


- (void)dispatchWithMethod:(NSString*)method
                      path:(NSString*)path
                parameters:(NSDictionary*)parameters
                   success:(void (^)())successBlock
                   faliure:(void (^)(BOOL shouldContinue))faliureBlock {
  
  NSLog(@"Dispatch event with debug dispatcher");
  
  NSLog(@"Request: \n%@\n%@", path, parameters);
  
  successBlock();
  
}


@end

File: PiwikTracker-Swift/PiwikDebugDispatcher.swift
//
//  PiwikDebugDispatcher.swift
//  PiwikTracker
//
//  Created by Cornelius Horstmann on 07.10.16.
//  Copyright © 2016 Mattias Levin. All rights reserved.
//

import Foundation

internal class PiwikDebugDispatcher: PiwikDispatcher {
    var userAgent: String? {
        didSet {
            print("Set custom user agent: \(userAgent)")
        }
    }
    
    func send(event: Event, success: ()->(), failure: (Bool)->()) {
        print("Request: \(event)")
        success()
    }
    
    func send(events: [Event], success: ()->(), failure: (Bool)->()) {
        for event in events {
            print("Request: \(event)")
        }
        success()
    }
}
